src/ProtocolHandler/HTTP/ChorizoHTTPConnectionHandler.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+using System.Text;
+using Chorizo.Logger;
+using Chorizo.ProtocolHandler.HTTP.Requests;
+using Chorizo.ProtocolHandler.HTTP.Responses;
+using Chorizo.ProtocolHandler.HTTP.Router;
+using Chorizo.Sockets.CzoSocket;
+
+namespace Chorizo.ProtocolHandler.HTTP
+{
+    public class ChorizoHTTPConnectionHandler:IChorizoProtocolConnectionHandler
+    {
+        private IRequestParser _rawRequestParser;
+        private IRouter _router;
+        private readonly IMiniLogger _optionalLogger;
+
+        public ChorizoHTTPConnectionHandler(
+            IMiniLogger optionalLogger = null,
+            IRequestParser rawRequestParser = null,
+            IRouter router = null
+        )
+        {
+            _router = router ?? new ChorizoRouter();
+            _rawRequestParser = rawRequestParser ?? new RawRequestParser();
+            _optionalLogger = optionalLogger;
+        }
+
+        public void HandleRequest(IChorizoSocket chorizoSocket)
+        {
+            var rawStartLineAndHeaders = receiveReqStartLineAndHeaders(chorizoSocket);
+            if (rawStartLineAndHeaders.Length <= 1)
+            {
+                chorizoSocket.Disconnect();
+            }
+            else
+            {
+                var parsedRequestInfo = _rawRequestParser.Parse(rawStartLineAndHeaders);
+                var contentLength = 0;
+                if (parsedRequestInfo.Headers.ContainsKey("CONTENT-LENGTH"))
+                {
+                    int.TryParse(parsedRequestInfo.Headers["CONTENT-LENGTH"], out contentLength);
+                }
+                var requestBody = receiveReqBody(chorizoSocket, contentLength);
+                var incomingRequest = new Request(parsedRequestInfo, requestBody);
```

Comment:
lines 38 - 44 seem to be parsing a request. It might make more sense to migrate that to `RawRequestParser`. I envision `.Parse` to return ` Request`